python-sdk/tests/test_config.py

```diff
@@ -0,0 +1,75 @@
+import unittest
+import pytest
+from tempfile import NamedTemporaryFile
+
+from pachyderm_sdk.config import ConfigFile
+from pachyderm_sdk.errors import ConfigError
+
+TEST_CONFIG = bytes("""
+{
+  "user_id": "some_user",
+  "v2": {
+    "active_context": "default",
+    "contexts": {
+      "default": {
+        "session_token": "token",
+        "cluster_deployment_id": "id",
+        "project": "default"
+      },
+      "other_context": {
+        "pachd_address": "grpc://localhost:12345",
+        "session_token": "other_token",
+        "cluster_deployment_id": "other_id",
+        "project": "other_project"
+      }
+    },
+    "metrics": true
+  }
+}
+""", 'utf-8')
+
+TEST_CONFIG_INVALID_CONTEXT = bytes("""
+{
+  "user_id": "some_user",
+  "v2": {
+    "active_context": "invalid_context",
+    "contexts": {
+      "default": {
+        "session_token": "token",
+        "cluster_deployment_id": "id",
+        "project": "default"
+      },
+      "other_context": {
+        "pachd_address": "grpc://localhost:12345",
+        "session_token": "other_token",
+        "cluster_deployment_id": "other_id",
+        "project": "other_project"
+      }
+    },
+    "metrics": true
+  }
+}
+""", 'utf-8')
+
+class TestConfig():
+    @staticmethod
+    def test_from_file():
```

Comment:
You can choose if you want to use this, but I opt to use tools that pytest provides. An example using this test would be:
```python
@staticmethod
def test_from_file(tmp_path):
    config_path = tmp_path / config.json
    config_path.write_bytes(TEST_CONFIG)
    c = ConfigFile(config_path)
    ...
```
(above code not tested)

The tmp_path gives you a `pathlib.Path` object to a temporary directory that gets cleaned-up after the test finishes. That way you don't need to use the `NamedTemporaryFile` and its somewhat weird api.